xdsl/utils/base_printer.py

```diff
@@ -68,6 +70,27 @@ def print_list(
                 self.print_string(delimiter)
             print_fn(elem)
 
+    @contextmanager
+    def delimited(self, delimiter: Delimiter):
```

Comment:
Could actually pass strings here instead, and avoid the delimiter API move?

Reply:
If the api move is just too nasty then I'll accept doing something else

Reply:
The `in_parens` and `in_braces` are from `StimPrinter`. Do you feel strongly about the helpers being in `Printer` instead of `BasePrinter`?

Reply:
I've implemented the changes (though left things in BasePrinter for now).

An interesting side effect is that you can now do things like `Printer.delimited("myattrname<", ">")`